### ISSUE TITLE
Request: pinned apps "launcher-behavior" on new workspace

### ISSUE THREAD

**Issue by Thomas-N** [robint99/mate-dock-applet#64]
Thanks for all the great work you've done on the dock. I have a feature request for a possible future update:

In the following situation:

- I have a pinned app, with an active window on workspace 1
- I'm currently on workspace 2
- I click the app icon

I'd like it to open a new window on workspace 2, instead of taking me back to workspace 1. 
Could this be introduced as optional behavior? Maybe also make the window-indicator dots only show up when there are active windows in the current workspace.

Neither ¨display pinned apps from all workspaces" or "display pinned apps only from current workspace" does exactly what I want.

Basically, I'd like pinned apps to always show up as launchers on all workspaces and only function as a "window list" when there are active windows on the current workspace.

**Comment by robint99** [comment]
The applet can already open a new instance of a running app rather than switching to an already open window. All you have to is hold down the `<shift>` key when clicking the app's icon and the new window will appear on whichever window is currently active. There's also an option to only display indicator dots for windows on the current workspace under the "Workspaces" tab on the preferences window.

For your other request, I can add an option for window lists to only show windows from the current workspace would an option to only display. If used with the display pinned apps from all workspaces option, this would seem to provide what you're after. But feel to correct if I'm wrong :)

**Comment by Thomas-N** [comment]
Thanks for your reply.

I think that would solve it, If I'm understanding correctly. What I'm looking for is a way to make the dock _never_ take me to a different workspace. For example, when I have a LibreOffice document opened in one workspace and another LibreOffice document in a second workspace, I always want the icon to take me to the version in my current workspace. Ideally, the "instance" of the dock in one workspace would act like it didn't even know what windows were active in other workspaces. (I still like to be able to launch my pinned apps from any workspace, though)

I don't see the option to display indicator dots only for current workspace. Looking at the changelog, I see that this was added in V0.71, and I'm using 0.7, as provided by Ubuntu Mate 16.04.1

**Comment by clessley** [comment]
Love the dock in Mutiny but I feel @Thomas-N 's pain. 

Full screen applications once clicked on from the dock _should_ move to the current workspace or have an option for the user to move it manually to a preferred workspace (right click, long click option, etc).

Compiz lets you do this from the left bullet window controls...but programs that run full screen can't be further moved once they begin because there are no windows controls or even borders to manipulate. Sometimes other apps that are also on the full screen program's workspace also move over once clicked on in the dock, overlaying it which is annoying. I can imagine wine apps/games doing this also.
